### ISSUE TITLE
feature request: have bard-cli read whole prompt (or a part of it) from stdin

### ISSUE THREAD

**Issue by kirisakow** [mosajjal/bard-cli#27]
I need bard-cli to handle input from stdin 

* Either as a rear part of the prompt, for example:

```sh
$ cat ./sometext.txt | bard-cli "recap the following text:"
```
<br/>

* Or as a complete substitute for the prompt, for example:
```sh
$ cat ./prompt.txt ./sometext.txt | bard-cli
```

**Comment by mosajjal** [comment]
I'm ok with adding the function. but it needs to look like this:

```
bard-cli -f "/path/to/file" "initial prompt before file" "another prompt before file"
```

for stdin, it should look like this:

```
cat /path/to/file" | bard-cli -f - "initial prompt" "another prompt" 
```

`-f` is crucial.

**Comment by kirisakow** [comment]
Sure! 

However, I'd like the default behavior (ie even with no value for `-f`) to be such that bard-cli always takes a look into stdin. In fact, this is actually [how the `cat` program works](https://www.man7.org/linux/man-pages/man1/cat.1.html):

```
SYNOPSIS
        cat [OPTION]... [FILE]...

DESCRIPTION
        Concatenate FILE(s) to standard output.

        With no FILE, or when FILE is -, read standard input.
```

What do you think?

**Comment by kirisakow** [comment]
So, here are the newly supported use cases:

```sh
# context: a side file
echo "2^10" > txt.txt
```

Read file contents from a side file:

```sh
# with a shorthand:
bard-cli -f ./txt.txt "compute the following:" 
# with a named argument:
bard-cli --file ./txt.txt "compute the following:" 
```

Read file contents from stdin:

```sh
# with a shorthand:
cat ./txt.txt | bard-cli -f - "compute the following:"
# with a named argument:
cat ./txt.txt | bard-cli --file - "compute the following:"
# with no argument or shorthand
cat ./txt.txt | bard-cli "compute the following:"
```

**Comment by mosajjal** [comment]
> ```shell
> # with no argument or shorthand
> cat ./txt.txt | bard-cli "compute the following:"
> ```

struggling to understand the use-case of this. if you want, you can always check if stdin is empty or not and alias `bard-cli` with `bard-cli -f -` yourself. doesn't need to be part of the code.

**Comment by mosajjal** [comment]
addressed by #28
